Assets/MRTK/Core/Providers/Hands/BaseHand.cs

```diff
@@ -78,6 +101,99 @@ protected void UpdateVelocity()
             frameOn++;
         }
 
+        protected bool IsInTeleportPose
```

Comment:
Doing this logic in BaseHand won't allow it to be used in classes which don't derive from BaseHand, like the WMRArticulatedHand. This feels like something that should be done in the ArticulatedHandDefinition, since it's API-agnostic and only depends on the finger joint data.

Reply:
Just saw your comment on

>Also the WMR articulated hand needs to be rewritten to derive from Basehand

This is one of the reasons the ArticulatedHandDefinition was created! It removes the need for a specific inheritance chain in the vein of "[composition over inheritance](https://en.wikipedia.org/wiki/Composition_over_inheritance)".

Reply:
Is there a reason that the WMRArticulatedHand doesn't derive from BaseHand? I'm ok with making the OculusHand use the ArticulatedHandDefinition, I just want to understand the broader design considerations.

Reply:
>Is there a reason that the WMRArticulatedHand doesn't derive from BaseHand?

The original inheritance of WMRArticulatedHand came from the APIs its data comes from. That API is shared among all WMR inputs (controllers, etc), so it made sense to share the API calls in a base class.

BaseHand started more from a similar place that ArticulatedHandDefinition did (API-agnostic but defines specific calculations and features of a hand) but was used differently (back to "composition over inheritance"). Instead of repurposing that class, I created a new one.

Overall, it was a miss on my part that these hand classes ever diverged so much in the first place, and ArticulatedHandDefinition was an attempt at reconciling that.